week3/project/ecommerce/src/hooks/useFavoriteItems.jsx

```diff
@@ -0,0 +1,28 @@
+import { useContext } from 'react'
+import { FavoriteIdItemContext } from '../contexts/FavoriteIdItemContext';
+import svgReg from '../images/heart-regular.svg'
+import svgSol from '../images/heart-solid.svg';
+
+export default function useFavoriteItems(item) {
+
+   const { idFavoriteItems, setIdFavoriteItems } = useContext(FavoriteIdItemContext);
+
+   let svgIcon = idFavoriteItems.includes(item) ? svgSol : svgReg;
+
+   function favoriteIdHandler(itemId) {
+      if (!idFavoriteItems.includes(itemId)) {
+         setIdFavoriteItems((prev) => [...prev, itemId])
+         svgIcon = svgSol;
+      }
+      else {
+         setIdFavoriteItems((prev) => prev.filter(id => id !== itemId))
+         svgIcon = svgReg
+      }
+   }
```

Comment:
and just for refactoring and it here also do you see that you used the condition of 
`idFavoriteItems.includes(item)`
to times and you did a parameter `ItemId` (again) and you already have it in line 6.

so yes it's great and totally works but after it works you can think about refactoring

so what about this version ?
```js
export default function useFavoriteItems(itemId) {
  const { idFavoriteItems, setIdFavoriteItems } = useContext(
    FavoriteIdItemContext
  );
  const isInFav = idFavoriteItems.includes(itemId);
  let svgIcon = isInFav ? svgSol : svgReg;

  function favoriteIdHandler() {
    if (!isInFav) {
      setIdFavoriteItems((prev) => [...prev, itemId]);
      svgIcon = svgSol;
    } else {
      setIdFavoriteItems((prev) => prev.filter((id) => id !== itemId));
      svgIcon = svgReg;
    }
  }

  return {
    svgIcon,
    favoriteIdHandler,
    idFavoriteItems,
  };
}
```

if you like it please don't copy it but try to do it yourself as practice. maybe the logic is too easy but refactoring is like a habit 

📝advise📝